### PR TITLE
Make pod tests pass, and run them from `make`

### PR DESCRIPTION
**Pull request by bhagany**
Problems with these tests were discovered while working on #566, so I figured I'd take a shot at fixing them.  I elected to change the code as little as possible, and change the tests instead, but I'm not super confident I made the right decision in every case.  Criticism welcome.

### CODE REVIEW

**Comment by bhagany** [comment]
Hrm, I take it Appveyor runs on Windows...

**Comment by arichiardi** [comment]
Great thanks!

**Comment by bhagany** [comment]
alright, my AppVeyor fix worked :)

**Comment by arichiardi** [comment]
Looks very good to me :)

File: boot/pod/test/boot/file_test.clj [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 (deftest relative-to-test
   (testing "Nil base"
-    (is (= "out/goog/base.js"          (str (relative-to nil      (io/file "out/goog/base.js"))))))
+    (is (thrown? AssertionError        (str (relative-to nil      (io/file "out/goog/base.js"))))))
```
Comment (bhagany) [inline boot/pod/test/boot/file_test.clj]: Yeah... `nil` is covered by a `:pre` condition in `relative-to`.  I thought about removing the test, but wasn't sure if there was a good reason for it to be there.  I think it's the only other option though, unless the `:pre` shouldn't be there.

File: boot/pod/src/boot/file.clj [hunk 2]
```diff
@@ -21,7 +21,7 @@
 (def ^:dynamic *sync-delete* true)
 (def ^:dynamic *hard-link*   true)
 
-(def windows? (boot.App/isWindows))
+(def ^:dynamic windows? (boot.App/isWindows))
```
Comment (bhagany) [inline boot/pod/src/boot/file.clj]: This is the change I was most unsure about.  The only reason for `windows?` to be dynamic is for testing purposes, and given that's the case, I didn't think that advertising its dynamic-ness with asterisks was a great idea.

At the time, I didn't know that CI ran on Windows as well, and that fact makes me think there's a better way to do this test? https://github.com/boot-clj/boot/pull/567/files#diff-b6e908691c9bd8d211924e12cfbbcd61R46

Comment (arichiardi) [inline boot/pod/src/boot/file.clj]: I mean it is just a minor thing anyways, given that we don't know all the implications here we might need to wait better opinion.

Comment (bhagany) [inline boot/pod/src/boot/file.clj]: Yeah, I agree.  I'm working on a better opinion myself :)

**Comment by bhagany** [comment]
Okay @arichiardi, I hope you'll like this.  It occurred to me that `io/file` will construct paths with platform-specific separators, so we can rely on AppVeyor tests to test Windows paths without doing it explicitly.  CI passed, so all is well!

I also removed the test for AssertionError, since the `:pre` check for `nil` on `relative-to` makes it redundant.  Let me know what you think.

**Comment by arichiardi** [comment]
Perfect! I think now it looks way way simpler. Maybe the original author of this ns can also chime in. In any case, good job!

**Comment by bhagany** [comment]
@arichiardi I just realized github is still saying that you're requesting changes.  Is that still true?  If so, I'm not sure what they are.

**Comment by arichiardi** [comment]
Uhm no, i am good, let me see if I can cancel the review.

**Comment by bhagany** [comment]
Fantastic, thanks :)

**Comment by alandipert** [comment]
:metal: :guitar: :metal:
